libraries/AP_Mount/AP_Mount_Xacti.cpp

```diff
@@ -0,0 +1,537 @@
+        return nullptr;
+    }
+
+    // search for backend with matching dronecan port and node id
+    for (uint8_t i = 0; i < ARRAY_SIZE(_detected_modules); i++) {
+        if (_detected_modules[i].driver != nullptr &&
+            _detected_modules[i].ap_dronecan == ap_dronecan &&
+            _detected_modules[i].node_id == node_id ) {
+            return _detected_modules[i].driver;
+        }
+    }
+
+    // if we got this far, this dronecan port and node id are not associated with any backend
+    // associate with first backend with node id of zero
+    for (uint8_t i = 0; i < ARRAY_SIZE(_detected_modules); i++) {
+        if (_detected_modules[i].driver != nullptr &&
+            _detected_modules[i].node_id == 0) {
+                _detected_modules[i].ap_dronecan = ap_dronecan;
+                _detected_modules[i].node_id = node_id;
+                const auto dev_id = AP_HAL::Device::make_bus_id(AP_HAL::Device::BUS_TYPE_UAVCAN,
+                                                                ap_dronecan->get_driver_index(),
+                                                                node_id, 0);
+                _detected_modules[i].driver->set_dev_id(dev_id);
+                return _detected_modules[i].driver;
+        }
+    }
+
+    return nullptr;
+}
+
+// handle xacti gimbal attitude status message
+void AP_Mount_Xacti::handle_gimbal_attitude_status(AP_DroneCAN* ap_dronecan, const CanardRxTransfer& transfer, const com_xacti_GimbalAttitudeStatus &msg)
+{
+    // fetch the matching backend driver, node id and gimbal id backend instance
+    AP_Mount_Xacti* driver = get_dronecan_backend(ap_dronecan, transfer.source_node_id);
+    if (driver == nullptr) {
+        return;
+    }
+
+    // convert body-frame Euler angles to Quaternion.  Note yaw direction is reversed from normal
+    driver->_current_attitude_quat.from_euler(radians(msg.gimbal_roll * 0.01), radians(msg.gimbal_pitch * 0.01), radians(-msg.gimbal_yaw * 0.01));
+    driver->_last_current_attitude_quat_ms = AP_HAL::millis();
+}
+
+// handle xacti gnss status request message
+void AP_Mount_Xacti::handle_gnss_status_req(AP_DroneCAN* ap_dronecan, const CanardRxTransfer& transfer, const com_xacti_GnssStatusReq &msg)
+{
+    // sanity check dronecan port
+    if (ap_dronecan == nullptr) {
+        return;
+    }
+
+    // get current location
+    uint8_t gps_status = 3;
+    Location loc;
+    if (!AP::ahrs().get_location(loc)) {
+        gps_status = 0;
+    }
+
+    // get UTC time
+    uint8_t hour, min, sec;
+    uint16_t ms;
+    if (!AP::rtc().get_local_time(hour, min, sec, ms)) {
+        hour = min = sec = ms = 0;
+    }
+
+    // send xacti specific gnss status message
+    com_xacti_GnssStatus xacti_gnss_status_msg {};
+    xacti_gnss_status_msg.gps_status = gps_status;
+    xacti_gnss_status_msg.order = msg.requirement;
+    xacti_gnss_status_msg.remain_buffer = 1;
+    xacti_gnss_status_msg.utc_year = 2023;      // ToDo: fix hardcoded year, month, day
```

Comment:
from MSP:
```
    struct tm* tm = gmtime(&utc_sec);

    p.year = tm->tm_year+1900;
    p.month = tm->tm_mon;
    p.day = tm->tm_mday;
    p.hour = tm->tm_hour;
    p.min = tm->tm_min;
    p.sec = tm->tm_sec;
```

Reply:
Great, thanks.  I've added a new method to RTC that looks like this

```
    // get date and time.  Returns true on success and fills in year, month, day, hour, min, sec and ms
    // year is the regular Gregorian year, month is 0~11, day is 1~31, hour is 0~23, minute is 0~59, second is 0~60 (1 leap second)
    bool get_date_and_time_utc(uint16_t& year, uint8_t& month, uint8_t& day, uint8_t &hour, uint8_t &min, uint8_t &sec) const;
```